nvme.c

```diff
@@ -5317,6 +5398,629 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
+	case NVME_REG_NSSR:
+		if (argconfig_parse_seen(opts, "nssr"))
+			nvme_set_register(bar, offset, cfg->nssr, cfg->mmio32);
+		break;
+	case NVME_REG_AQA:
+		if (argconfig_parse_seen(opts, "aqa"))
+			nvme_set_register(bar, offset, cfg->aqa, cfg->mmio32);
+		break;
+	case NVME_REG_ASQ:
+		if (argconfig_parse_seen(opts, "asq"))
+			nvme_set_register(bar, offset, cfg->asq, cfg->mmio32);
+		break;
+	case NVME_REG_ACQ:
+		if (argconfig_parse_seen(opts, "acq"))
+			nvme_set_register(bar, offset, cfg->acq, cfg->mmio32);
+		break;
+	case NVME_REG_BPRSEL:
+		if (argconfig_parse_seen(opts, "bprsel"))
+			nvme_set_register(bar, offset, cfg->bprsel, cfg->mmio32);
+		break;
+	case NVME_REG_CMBMSC:
+		if (argconfig_parse_seen(opts, "cmbmsc"))
+			nvme_set_register(bar, offset, cfg->cmbmsc, cfg->mmio32);
+		break;
+	case NVME_REG_NSSD:
+		if (argconfig_parse_seen(opts, "nssd"))
+			nvme_set_register(bar, offset, cfg->nssd, cfg->mmio32);
+		break;
+	case NVME_REG_PMRCTL:
+		if (argconfig_parse_seen(opts, "pmrctl"))
+			nvme_set_register(bar, offset, cfg->pmrctl, cfg->mmio32);
+		break;
+	case NVME_REG_PMRMSCL:
+		if (argconfig_parse_seen(opts, "pmrmscl"))
+			nvme_set_register(bar, offset, cfg->pmrmscl, cfg->mmio32);
+		break;
+	case NVME_REG_PMRMSCU:
+		if (argconfig_parse_seen(opts, "pmrmscu"))
+			nvme_set_register(bar, offset, cfg->pmrmscu, cfg->mmio32);
+		break;
+	default:
+		break;
+	}
+}
+
+static int set_register_check(struct argconfig_commandline_options *opts,
+			      struct set_reg_config *cfg, int offset)
+{
+	bool flag = false;
+
+	switch (offset) {
+	case NVME_REG_INTMS:
+		if (argconfig_parse_seen(opts, "intms"))
+			flag = true;
+		break;
+	case NVME_REG_INTMC:
+		if (argconfig_parse_seen(opts, "intmc"))
+			flag = true;
+		break;
+	case NVME_REG_CC:
+		if (argconfig_parse_seen(opts, "cc"))
+			flag = true;
+		break;
+	case NVME_REG_CSTS:
+		if (argconfig_parse_seen(opts, "csts"))
+			flag = true;
+		break;
+	case NVME_REG_NSSR:
+		if (argconfig_parse_seen(opts, "nssr"))
+			flag = true;
+		break;
+	case NVME_REG_AQA:
+		if (argconfig_parse_seen(opts, "aqa"))
+			flag = true;
+		break;
+	case NVME_REG_ASQ:
+		if (argconfig_parse_seen(opts, "asq"))
+			flag = true;
+		break;
+	case NVME_REG_ACQ:
+		if (argconfig_parse_seen(opts, "acq"))
+			flag = true;
+		break;
+	case NVME_REG_BPRSEL:
+		if (argconfig_parse_seen(opts, "bprsel"))
+			flag = true;
+		break;
+	case NVME_REG_CMBMSC:
+		if (argconfig_parse_seen(opts, "cmbmsc"))
+			flag = true;
+		break;
+	case NVME_REG_NSSD:
+		if (argconfig_parse_seen(opts, "nssd"))
+			flag = true;
+		break;
+	case NVME_REG_PMRCTL:
+		if (argconfig_parse_seen(opts, "pmrctl"))
+			flag = true;
+		break;
+	case NVME_REG_PMRMSCL:
+		if (argconfig_parse_seen(opts, "pmrmscl"))
+			flag = true;
+		break;
+	case NVME_REG_PMRMSCU:
+		if (argconfig_parse_seen(opts, "pmrmscu"))
+			flag = true;
+		break;
+	default:
+		break;
+	}
+
+	if (flag && cfg->offset == offset) {
+		nvme_show_error("offset duplicated param");
```

Comment:
This won't detect multiple flag options being specified, e.g. `--cc --csts`. Nor will it detect `--offset` with an unrecognized offset value being specified alongside a flag argument.

Reply:
The set-reg command named arguemnts are needed values also to set as `--cc=<NUM> --csts=<NUM>`. Also the command `--offset` arguemnt value is not checked as if an unrecognized offset value or not.